clang/test/Driver/cl-cxx20-modules.cppm

```diff
@@ -14,3 +14,11 @@
 
 //--- test.pcm
 // CPP20WARNING-NOT: clang-cl: warning: argument unused during compilation: '/std:c++20' [-Wunused-command-line-argument]
+
+// test whether the following outputs %Hello.bmi
+// RUN: %clang_cl /std:c++20 --precompile -x c++-module -fmodule-output=%t/Hello.bmi -Fo"%t/Hello.bmi" -c %t/Hello.cppm -### 2>&1 | FileCheck %s
```

Comment:
This is just one invocation, but the logic in GetNamedOutputPath() seems to cover many cases.

Also, is it `-fmodule-output` or `/Fo` that sets the output filename (or both?)

(Nit: you'll want to put `--` before `%t/Hello.cppm`, otherwise it may be interpreted as a command-line option e.g. if the path starts with `/Users`.)

Reply:
> This is just one invocation, but the logic in GetNamedOutputPath() seems to cover many cases.

Agreed; once again; open to suggestions for more test cases. 

> Also, is it -fmodule-output or /Fo that sets the output filename (or both?)

From the [Clang user guide](https://clang.llvm.org/docs/StandardCPlusPlusModules.html#how-to-build-projects-using-modules), it appears _both_ are required, and CMake in particular uses the 'one-phase compilation' method.

Reply:
From the doc:

> To generate a BMI for an importable module unit, use either the --precompile or -fmodule-output command line options.

Your test uses *both*, with the same filename passed to two different flags. This seems wrong to me.

Reply:
Coming back after a long while...

Both arguments were provided to exercise [CMake's command-line invocation](https://gitlab.kitware.com/cmake/cmake/-/issues/25731#note_1603044) which given this information might be incorrect.

@ChuanqiXu9 are you able to provide more information?

Reply:
Sorry for ignoring this. `--precompile` will be meaningless if it is followed by a `-c`. So the command line in cmake's thread actually use `-fmodule-output`.

Reply:
Please don't worry too much about what CMake's command-line invocations [linked above](https://github.com/llvm/llvm-project/pull/121046#discussion_r2038627647) are doing.  CMake doesn't yet support modules with `clang-cl`, and is waiting on the work here.  The command lines I posted over there are showing what *unmerged* changes to CMake are doing in its test suite as part of analyzing why not everything is working.  This PR should focus on the functionality in `clang-cl` and its test should focus on the documented and intended usage.  Once this is merged then CMake can be taught to do the right thing.